### PR TITLE
Jw salarybalance

### PR DESCRIPTION
**Pull request by Nandhinidevi123**
salary balance fetch from last salary slip or inside employee salary balance

### CODE REVIEW

File: ganapathy_pavers/ganapathy_pavers/report/job_worker___report/job_worker___report.py [hunk 1]
```diff
@@ -144,3 +154,16 @@ def get_columns(other_work):
 		]
 	
 	return columns
+
+def get_employee_salary_balance(employee, from_date, to_date):
+   
+    salary_slip=frappe.db.sql("""select total_unpaid_amount from `tabSalary Slip` where posting_date between %(from)s and %(to)s and employee=%(emp)s order by posting_date desc,modified desc limit 1""",{"emp":employee, "from":from_date, "to":to_date},as_dict=True)
+    salary_slip1=frappe.db.sql("""select total_unpaid_amount from `tabSalary Slip` where employee=%(emp)s order by posting_date desc,modified desc limit 1""",{"emp":employee},as_dict=True)
+    salary_slip2=frappe.db.sql("""select salary_balance from `tabEmployee` where name=%(emp)s and status="Active """,{"emp":employee},as_dict=True)
```
Comment (Ponnusamy1-V) [inline ganapathy_pavers/ganapathy_pavers/report/job_worker___report/job_worker___report.py:162]: add this condition also `docstatus=1`

File: ganapathy_pavers/ganapathy_pavers/report/job_worker___report/job_worker___report.py [hunk 2]
```diff
@@ -144,3 +154,16 @@ def get_columns(other_work):
 		]
 	
 	return columns
+
+def get_employee_salary_balance(employee, from_date, to_date):
+   
+    salary_slip=frappe.db.sql("""select total_unpaid_amount from `tabSalary Slip` where posting_date between %(from)s and %(to)s and employee=%(emp)s order by posting_date desc,modified desc limit 1""",{"emp":employee, "from":from_date, "to":to_date},as_dict=True)
+    salary_slip1=frappe.db.sql("""select total_unpaid_amount from `tabSalary Slip` where employee=%(emp)s order by posting_date desc,modified desc limit 1""",{"emp":employee},as_dict=True)
```
Comment (Ponnusamy1-V) [inline ganapathy_pavers/ganapathy_pavers/report/job_worker___report/job_worker___report.py:161]: In this query, get last doc before the `from_date`

**Review by Ponnusamy1-V — CHANGES_REQUESTED** [review]

File: ganapathy_pavers/ganapathy_pavers/report/job_worker___report/job_worker___report.py (hunk 1 above)
Comment (Nandhinidevi123) [reply ganapathy_pavers/ganapathy_pavers/report/job_worker___report/job_worker___report.py]: resolved

File: ganapathy_pavers/ganapathy_pavers/report/job_worker___report/job_worker___report.py (hunk 2 above)
Comment (Nandhinidevi123) [reply ganapathy_pavers/ganapathy_pavers/report/job_worker___report/job_worker___report.py]: fixed

**Pull request merged**